provider/common/bootstrap.go

```diff
@@ -180,11 +181,41 @@ func BootstrapInstance(ctx environs.BootstrapContext, env environs.Environ, args
 		ImageMetadata:   imageMetadata,
 		StatusCallback:  instanceStatus,
 		CleanupCallback: statusCleanup,
-	})
-	if err != nil {
+	}
+	zones, err := startInstanceZones(env, startInstanceArgs)
+	if errors.IsNotImplemented(err) {
+		// No zone support, so just call StartInstance with
+		// a blank StartInstanceParams.AvailabilityZone.
+		zones = []string{""}
```

Comment:
Is it better to start with a list containing a single empty entry rather than an empty list? The latter sounds much more like "pick anything" than the former (to me).

Reply:
Any empty list wouldn't work, since we're iterating over the list...